presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/CachingJdbcClient.java

```diff
@@ -196,7 +196,7 @@ public boolean isLimitGuaranteed(ConnectorSession session)
             return cachedTableHandle;
         }
         Optional<JdbcTableHandle> tableHandle = delegate.getTableHandle(identity, schemaTableName);
-        if (tableHandle.isEmpty() || cacheMissing) {
+        if (tableHandle.isPresent() || cacheMissing) {
```

Comment:
Can you please add a test for that?

Reply:
maybe no-one needed this cache after all :)

Reply:
@kokosing as discussed, test will be delivered in separate pr. Please AC.

Reply:
test is added as separate commit in this pr.